ui/app/models/transform.js

```diff
@@ -0,0 +1,91 @@
+import { alias } from '@ember/object/computed';
+import { computed } from '@ember/object';
+import DS from 'ember-data';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+
+const { attr } = DS;
+
+// these arrays define the order in which the fields will be displayed
+// see
+//https://www.vaultproject.io/api-docs/secret/transform#create-update-transformation
+const TYPES = [
+  {
+    value: 'fpe',
+    displayName: 'Format Preserving Encryption (FPE)',
+  },
+  {
+    value: 'masking',
+    displayName: 'Masking',
+  },
+];
+
+const TWEAK_SOURCE = [
+  {
+    value: 'supplied',
+    displayName: 'supplied',
+  },
+  {
+    value: 'generated',
+    displayName: 'generated',
+  },
+  {
+    value: 'internal',
+    displayName: 'internal',
+  },
+];
+
+export default DS.Model.extend({
+  // TODO: for now, commenting out openApi info, but keeping here just in case we end up using it.
+  // useOpenAPI: true,
```

Comment:
you might've already explored this, but juuust in case: it's possible to combine the attrs from openApi with the ones defined in the model ([example here](https://github.com/hashicorp/vault/blob/24fad4f96177906150a1e417ce35e4ea3fa5547d/ui/app/models/auth-config/github.js#L16-L24)). this also allows you to separate attrs into field groups and control the order since they're rendered in the same order as the array (i think, not totally sure). this would give you the benefit of not having to update the model if/when a backend attribute changes while still giving you the flexibility of ultimately rendering the `transform-edit-component` you've built below. a bit of a hybrid approach, i suppose, but worth considering!